cluster-autoscaler/cloudprovider/packet/packet_cloud_provider.go

```diff
@@ -164,9 +175,7 @@ func BuildPacket(opts config.AutoscalingOptions, do cloudprovider.NodeGroupDisco
 		klog.Fatalf("Must specify at least one node group with --nodes=<min>:<max>:<name>,...")
 	}
 
-	if len(do.NodeGroupSpecs) > 1 {
-		klog.Fatalf("Packet autoscaler only supports a single nodegroup for now")
-	}
+	validNodepoolName := regexp.MustCompile(`^[a-z0-9A-Z]+[a-z0-9A-Z\-\.\_]*[a-z0-9A-Z]+$|^[a-z0-9A-Z]{1}$`)
```

Comment:
I have no argument with validating the nodepool name, but why is this inside the packet provider? Is this not handled at a higher level?

Reply:
AFAIK this is not handed at a higher level.

Reply:
I guess it needs to stay here then.